src/common/state/store.ts

```diff
@@ -1,13 +1,34 @@
-import { configureStore } from '@reduxjs/toolkit';
+import { combineReducers, configureStore, EnhancedStore } from '@reduxjs/toolkit';
+import { networkSlice } from './network-slice';
 import { modalSlice } from '@components/modals/modal-slice';
 import { searchSlice } from '@features/search/search-slice';
+import { nextReduxCookieMiddleware, wrapMakeStore } from 'next-redux-cookie-wrapper';
+import { createWrapper } from 'next-redux-wrapper';
 
-export const store = configureStore({
-  reducer: {
-    modal: modalSlice.reducer,
-    search: searchSlice.reducer,
-  },
+const rootReducer = combineReducers({
+  global: networkSlice.reducer,
+  modal: modalSlice.reducer,
+  search: searchSlice.reducer,
 });
 
-export type RootState = ReturnType<typeof store.getState>;
-export type AppDispatch = typeof store.dispatch;
+export let store: EnhancedStore;
+const makeStore = wrapMakeStore(() => {
```

Comment:
Couldn't use `atomWithStore` here because the store is being created inside the wrapper. Maybe there's another way that I'm not aware of.